
\acrodef{fpr}[FPR]{false positive rate}
\acrodef{lot}[LOT]{largely orthologous tree}
\acrodef{mpl}[MPL]{mean path length}
\acrodef{mgp}[MGP]{Mammalian Genome Project}
\acrodef{tc}[TC]{taxonomic coverage}
\acrodef{tcc}[TCC]{taxonomic coverage constraint}
\acrodef{slr}[SLR]{Sitewise Likelihood Ratio}
%\acrodef{lr}{LR}{length ratio}
%\acrodef{wcs}{WCS}{windows of clustered substitution}
\acrodef{ml}[ML]{maximum likelihood}
\acrodef{psc}[PSC]{positively selected codon}
\acrodef{psg}[PSG]{positively selected gene}
\acrodef{psfive}[PSC$_{5\%}$]{positively selected codon}
\acrodef{dfe}[DFE]{distribution of fitness effects}
\acrodef{lrt}[LRT]{likelihood ratio test}
\acrodef{lr}[LR]{likelihood ratio}
\acrodef{fwer}[FWER]{family-wise error rate}
\acrodef{fdr}[FDR]{false discovery rate}
\acrodef{tpm}[TPM]{truncated product method}
\acrodef{go}[GO]{Gene Ontology}
\acrodef{fet}[FET]{Fisher's Exact Test}
\acrodef{slc}[SLC]{solute carrier}
\acrodef{roc}[ROC]{receiver operator characteristic}
\acrodef{npp}[NPP]{nearby paralog pair}
\acrodef{mya}[mya]{million years ago}
\acrodef{myr}[Myr]{million years}
\acrodef{ne}[$N_{e}$]{effective population size}
\acrodef{2r}[2R]{two round}
\acrodef{ils}[ILS]{incomplete lineage sorting}
\acrodef{ils}[ILS]{incomplete lineage sorting}
\acrodef{paml}[PAML]{Phylogenetic Analysis by Maximum Likelihood}

%\newcommand{\tocite}[2]{\draft{To cite: \citet{#1} --- #2}}
\newcommand{\tocite}[2]{}

\newcommand{\TODO}[1]{\textcolor{red}{TODO} \todo{#1}\xspace}
\newcommand{\draft}[1]{\textcolor{gray}{[#1]}\xspace}
%\newcommand{\todo}{\textcolor{red}{TO-DO}\xspace}

\newcommand{\gene}[1]{\emph{#1}}
\newcommand{\species}[1]{\emph{#1}}

% Intro
\newcommand{\syn}{synonymous\xspace}
\newcommand{\nsyn}{nonsynonymous\xspace}

\newcommand{\sw}{sitewise\xspace}
\newcommand{\ml}{maximum likelihood\xspace}

\newcommand{\slr}{SLR\xspace}
\newcommand{\pamlTwo}{PAML M2\xspace}
\newcommand{\pamlEight}{PAML M8\xspace}

\newcommand{\fpr}{FPR\xspace}
\newcommand{\pss}{PSS\xspace}
\newcommand{\psg}{PSG\xspace}

% Indel1
\newcommand{\mtwo}{PAML M2a\xspace}
\newcommand{\meight}{PAML M8\xspace}
\newcommand{\pranka}{PRANK$_{\textrm{AA}}$\xspace}
\newcommand{\prankc}{PRANK$_{\textrm{C}}$\xspace}
\newcommand{\tpr}{TPR$_{1\%}$\xspace}
\newcommand{\tprf}{TPR$_{5\%}$\xspace}
\newcommand{\Sw}{Sitewise\xspace}
\newcommand{\Dr}{{\emph{Drosophila}}\xspace}
%\newcommand{\tcr}[1]{\textcolor{red}{#1}}
%\newcommand{\tcrr}[1]{\textcolor{red}{#1}}
\newcommand{\tcr}[1]{#1}
\newcommand{\tcrr}[1]{#1}
\newcommand{\nh}{non-homologous\xspace}
\newcommand{\hidegraphics}[2]{\includegraphics[#1]{#2}}
%\newcommand{\hidegraphics}[2]{}


% Orthologs
\newcommand{\ens}{Ensembl\xspace}
\newcommand{\cmp}{Compara\xspace}
\newcommand{\lcv}{low-coverage\xspace}
\newcommand{\euth}{Eutherian\xspace}
\newcommand{\mammln}{mammalian\xspace}
\newcommand{\mamml}{mammal\xspace}
\newcommand{\zcop}{zero-copy\xspace}

% Mammals1

\newcommand{\mgp}{Mammalian Genome Project\xspace}
\newcommand{\hclust}{$hcluster\_sg$\xspace}
\newcommand{\subtr}{subtree\xspace}
\newcommand{\prank}{PRANK\xspace}

\newcommand{\ngenes}{15,XYZ\xspace}
\newcommand{\dnds}{$d$N$/$$d$S\xspace}
\newcommand{\dn}{$d$N\xspace}
\newcommand{\ds}{$d$S\xspace}
\newcommand{\kpa}{$\kappa$\xspace}
\newcommand{\omg}{$\omega$\xspace}
\newcommand{\omgml}{$\omega_{ML}$\xspace}
\newcommand{\omgmean}{$\bar{\omega}$\xspace}
\newcommand{\psc}{PSC\xspace}
\newcommand{\nsc}{NSC\xspace}
\newcommand{\nsfive}{NSC$_{5\%}$\xspace}
\newcommand{\psfive}{PSC$_{5\%}$\xspace}
\newcommand{\psten}{PSC$_{10\%}$\xspace}
\newcommand{\nsten}{NSC$_{10\%}$\xspace}
\newcommand{\psone}{PSC$_{1\%}$\xspace}
\newcommand{\psfdr}{PSC$_{FDR}$\xspace}

\newcommand{\ngap}{non-gap\xspace}
\newcommand{\Ngap}{Non-gap\xspace}

\newcommand{\fpos}{$F_{pos}$\xspace}
\newcommand{\fblw}{$F_{<1}$\xspace}
\newcommand{\fabv}{$F_{>1}$\xspace}

\newcommand{\mpl}{MPL\xspace}

\newcommand{\confint}{confidence interval\xspace}

\newcommand{\pos}[1]{Pos$_{#1}$\xspace}
\newcommand{\chisqlt}[1]{$P_{\chi^{2}_{1}}<#1$\xspace}
\newcommand{\chisq}{$\chi^{2}_{1}$\xspace}
\newcommand{\bhfdr}[1]{FDR$<#1$\xspace}
\newcommand{\slrt}{LRT$_{SLR}$\xspace}
\newcommand{\ci}{CI$_{95\%}$\xspace}
\newcommand{\ciup}{CI$_{upper}$\xspace}
\newcommand{\cidown}{CI$_{lower}$\xspace}

\newcommand{\nz}{nonzero\xspace}

\newcommand{\Ne}{N$_{E}$\xspace}

\newcommand{\nhom}{non-homologous\xspace}


% Mammals 2

\newcommand{\psghoch}{Hochberg\xspace}

\newcommand{\psgtfive}{$TPM_{0.05}$\xspace}
\newcommand{\psgtten}{$TPM_{0.1}$\xspace}
\newcommand{\psgttwenty}{$TPM_{0.2}$\xspace}
\newcommand{\psgtfifty}{$TPM_{0.5}$\xspace}

\newcommand{\psgfisher}{$Fis.$\xspace}

\newcommand{\psgeohfive}{$Emp_{0.005}$\xspace}
\newcommand{\psgeone}{$Emp_{0.01}$\xspace}
\newcommand{\psgefive}{$Emp_{0.05}$\xspace}
\newcommand{\psgeten}{$Emp_{0.1}$\xspace}

\newcommand{\wa}{$\bar{\omega}_{A}$\xspace}
\newcommand{\wg}{$\bar{\omega}_{G}$\xspace}

\newcommand{\topgo}{\texttt{topGO}\xspace}

\newcommand{\bbtable}{\begin{landscape} \begin{table}}
\newcommand{\eetable}{\end{table} \end{landscape}}
%\newcommand{\bbtable}{\begin{sidewaystable}}
%\newcommand{\eetable}{\end{sidewaystable}}

%\newcommand{\bbfig}{\begin{landscape} \begin{figure}}
%\newcommand{\eefig}{\end{figure} \end{landscape}}
\newcommand{\bbfig}{\begin{sidewaysfigure}}
\newcommand{\eefig}{\end{sidewaysfigure}}
