\begin{center}
\Large \mytitle

\large Summary
\end{center}
\small
\hfill Gregory Jordan \\
\noindent{\today} \hfill Darwin College \\
\small
Insight into the evolution of protein-coding genes can be gained from
the use of phylogenetic codon models. Recently sequenced mammalian
genomes and powerful analysis methods developed over the past decade
provide the potential to globally measure the impact of natural
selection on protein sequences at a fine scale. The detection of
positive selection in particular is of great interest, with relevance
to the study of host-parasite conflicts, immune system evolution and
adaptive differences between species. This thesis examines the
performance of methods for detecting positive selection first with a
series of simulation experiments, and then with two empirical studies
in mammals and primates.

Our ability to make confident estimates of the prevalence of positive
selection in proteins has been hampered to some extent by uncertainty
regarding the level of false positives resulting from alignment
error. To this end, I conduct a simulation study to estimate the rate
of false positive results attributable to alignment error. A variety
of aligners and alignment filtering methods are compared, showing a
striking difference between aligners in their tendency to produce
false positive results. Under most conditions, the best aligners tend
to produce very few false positives due to misalignment.

The rest of this thesis focuses on two genome-wide studies: an
analysis of sitewise selective pressures across 38 mammalian genomes,
and a genome-wide scan for genes with evidence of accelerated
evolution in gorilla and the African great apes. In the broader
mammalian analysis, the global distribution of sitewise evolutionary
constraint is characterized and strong evidence is presented for less
positive selection in the protein-coding genes of rodents compared to
primates and other mammalian orders. New methods are developed for
combining sitewise estimates across genes and protein-coding domains,
revealing widespread signals of positive selection in genes and
domains related to host defense and, surprisingly, centromere
binding. The African great ape analysis uses phylogenetic codon models
to identify genes which have experienced elevated evolutionary rates
in gorilla, human and chimpanzee. Similar numbers of accelerated genes
are identified in each of these genomes, and several accelerated genes
are identified in gorilla with plausible relationships to its unique
phenotypic and behavioral characteristics. Finally, genome-wide coding
alignments are used to infer genome-wide selective pressures along
each branch of the great ape tree, providing corroborating evidence of
a trend towards decreasing population sizes in the recent evolutionary
history of African great apes.
