\section*{Acronyms}
\label{Acronyms}
\begin{acronym}[TDMA]

\acro{fpr}[FPR]{false positive rate}
\acro{lot}[LOT]{largely orthologous tree}
\acro{mpl}[MPL]{mean path length}
\acro{mgp}[MGP]{Mammalian Genome Project}
\acro{tc}[TC]{taxonomic coverage}
\acro{tcc}[TCC]{taxonomic coverage constraint}
\acro{slr}[SLR]{Sitewise Likelihood Ratio}
\acro{wcs}{WCS}{windows of clustered substitution}
\acro{ml}[ML]{maximum likelihood}
\acro{mle}[MLE]{maximum likelihood estimate}
\acro{psc}[PSC]{positively selected codon}
\acro{psg}[PSG]{positively selected gene}
\acro{dfe}[DFE]{distribution of fitness effects}
\acro{lrt}[LRT]{likelihood ratio test}
\acro{lr}[LR]{likelihood ratio}
\acro{fwer}[FWER]{family-wise error rate}
\acro{fdr}[FDR]{false discovery rate}
\acro{tpm}[TPM]{truncated product method}
\acro{go}[GO]{Gene Ontology}
\acro{fet}[FET]{Fisher's Exact Test}
\acro{slc}[SLC]{solute carrier}
\acro{roc}[ROC]{receiver operator characteristic}
\acro{npp}[NPP]{nearby paralog pair}
\acro{mya}[mya]{million years ago}
\acro{myr}[Myr]{million years}
\acro{ne}[$N_{e}$]{effective population size}
\acro{2rgd}[2R]{two rounds of genome duplication}
\acro{2r}[2R]{two rounds}
\acro{ils}[ILS]{incomplete lineage sorting}
\acro{paml}[PAML]{Phylogenetic Analysis by Maximum Likelihood}
\acro{encode}[ENCODE]{Encyclopedia of DNA Elements}
\acro{optic}[OPTIC]{Orthologous and Paralogous Transcripts in Clades}
\acro{mbl2}[\emph{MBL2}]{mannose-binding lectin 2}
\acro{masp}[MASP]{MBL-associated serine protease}
\acro{tlr}[TLR]{Toll-like receptor}
\acro{hcg}[HCG]{human-chimpanzee-gorilla}
\acro{hc}[HC]{human-chimpanzee}
\acro{gh}[GH]{gorila and human}
\acro{gc}[GC]{gorila and chimpanzee}
\acro{ch}[CH]{chimpanzee and human}
\acro{aga}[AGA]{African great ape}
\acro{epo}[EPO]{Enredo-Pecan-Ortheus}
\acro{pwf}[PWF]{probability weighting function}
\acro{wd}[WD]{working draft}

\end{acronym}
